implementations/rust/message/src/message.rs

```diff
@@ -45,39 +44,15 @@ pub mod message {
     }
 
     impl Codec for Message {
-        type Inner = Message;
-        fn encode(msg: &Message, u: &mut Vec<u8>) -> Result<(), String> {
-            Route::encode(&msg.onward_route, u);
-            Route::encode(&msg.return_route, u);
+        type Inner = Box<Message>;
+        fn encode(msg: Box<Message>, u: &mut Vec<u8>) -> Result<(), String> {
```

Comment:
```suggestion
        fn encode(msg: Box<Message>, u: &mut [u8]) -> Result<(), String> {
```

Reply:
It's a Vec because its size is unknown up-front, it needs to be growable.